### DISCUSSION TITLE
Dropdown in Classes with Pre populated list of classes

### DISCUSSION THREAD

**Discussion by ahmadlimbada** [GrapesJS/grapesjs#4425] [General]
Currently each and every time, user need to write class name. So is there anyway we can show the list of existing classes in that class manager while adding the class.

I know we can get classes list from Style Manager however mainly class name editor needs to be overwritten. @artf kindly let me know if any references are there for overriding the following part. So instead of writing class name every time I just want to show the list of existing class over here. 
![Screenshot (24)](https://user-images.githubusercontent.com/30268583/177349154-f52c09f9-1f41-46da-b3a3-5b9a806555c4.png)

**Comment by lexoyo** [comment]
Hi
I recently did this, feel free to try it
https://www.npmjs.com/package/grapesjs-ui-suggest-classes

**Reply by ahmadlimbada** [reply]
@lexoyo Thanks a lot! I'll surely look into it.

**Reply by ahmadlimbada** [reply]
@lexoyo I went through your code and I just had some suggestion. Instead of updating every 'selector' event it can be changed to 'open-sm' so it does not slow down gjs when number of components are more on the canvas. While using your code I noticed lag when there are more components on the canvas and second thing is that we can reduce number of classes to top 5 used classes otherwise if more classes are there it might go out of window. If you want I can create MR for the same. btw Thanks for your help Mate:)

**Reply by lexoyo** [reply]
Thx for the feedback

I sure would appreciate a MR, i didn't use it in production yet

Also for the lag, your suggestion makes sense, I'll do that before I use it in silex, 🙏
